packages/block-library/src/image/view.js

```diff
@@ -77,269 +77,278 @@ function handleScroll( ctx ) {
+					event.preventDefault();
+				}
+			},
+			handleTouchEnd() {
+				// We need to wait a few milliseconds before resetting
+				// to ensure that pinch to zoom works consistently
+				// on mobile devices when the lightbox is open.
+				lastTouchTime = Date.now();
+				isTouching = false;
+			},
 		},
-		setButtonStyles() {
-			const { ref } = getElement();
-			const { naturalWidth, naturalHeight, offsetWidth, offsetHeight } =
-				ref;
-
-			// If the image isn't loaded yet, we can't
-			// calculate where the button should be.
-			if ( naturalWidth === 0 || naturalHeight === 0 ) {
-				return;
-			}
+		callbacks: {
+			initOriginImage() {
+				const ctx = getContext();
+				const { ref } = getElement();
+				ctx.imageRef = ref;
+				if ( ref.complete ) {
+					ctx.imageLoaded = true;
+					ctx.imageCurrentSrc = ref.currentSrc;
+				}
+			},
+			initTriggerButton() {
+				const ctx = getContext();
+				const { ref } = getElement();
+				ctx.lightboxTriggerRef = ref;
+			},
+			initLightbox() {
+				const ctx = getContext();
+				const { ref } = getElement();
+				if ( ctx.lightboxEnabled ) {
+					const focusableElements =
+						ref.querySelectorAll( focusableSelectors );
+					ctx.firstFocusableElement = focusableElements[ 0 ];
+					ctx.lastFocusableElement =
+						focusableElements[ focusableElements.length - 1 ];
+
+					// Move focus to the dialog when opening it.
+					ref.focus();
+				}
+			},
+			setButtonStyles() {
+				const { ref } = getElement();
+				const {
+					naturalWidth,
+					naturalHeight,
+					offsetWidth,
+					offsetHeight,
+				} = ref;
+
+				// If the image isn't loaded yet, we can't
+				// calculate where the button should be.
+				if ( naturalWidth === 0 || naturalHeight === 0 ) {
+					return;
+				}
 
-			const figure = ref.parentElement;
-			const figureWidth = ref.parentElement.clientWidth;
-
-			// We need special handling for the height because
-			// a caption will cause the figure to be taller than
-			// the image, which means we need to account for that
-			// when calculating the placement of the button in the
-			// top right corner of the image.
-			let figureHeight = ref.parentElement.clientHeight;
-			const caption = figure.querySelector( 'figcaption' );
-			if ( caption ) {
-				const captionComputedStyle = window.getComputedStyle( caption );
-				if (
-					! [ 'absolute', 'fixed' ].includes(
-						captionComputedStyle.position
-					)
-				) {
-					figureHeight =
-						figureHeight -
-						caption.offsetHeight -
-						parseFloat( captionComputedStyle.marginTop ) -
-						parseFloat( captionComputedStyle.marginBottom );
+				const figure = ref.parentElement;
+				const figureWidth = ref.parentElement.clientWidth;
+
+				// We need special handling for the height because
+				// a caption will cause the figure to be taller than
+				// the image, which means we need to account for that
+				// when calculating the placement of the button in the
+				// top right corner of the image.
```

Comment:
```suggestion
				/*
				 * A special handling for the height is needed because
				 * a caption will cause the figure to be taller than
				 * the image, which means is necessary
				 * when calculating the placement of the button in the
				 * top right corner of the image.
				 */
```